### PR TITLE
PolicyInsights: Support querying policy evaluation details

### PR DESCRIPTION
**Pull request by j5lim**
## Description

Add support for querying policy evaluation details of policyStates.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/PolicyInsights/PolicyInsights.Test/SessionRecords/Microsoft.Azure.Commands.PolicyInsights.Test.ScenarioTests.PolicyStateSummaryTests/ManagementGroupScope.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "Entries": [
     {
-      "RequestUri": "/providers/Microsoft.Management/managementGroups/azgovtest4/providers/Microsoft.PolicyInsights/policyStates/latest/summarize?api-version=2018-04-04&$top=10&$from=2018-03-31T00:00:00Z",
-      "EncodedRequestUri": "L3Byb3ZpZGVycy9NaWNyb3NvZnQuTWFuYWdlbWVudC9tYW5hZ2VtZW50R3JvdXBzL2F6Z292dGVzdDQvcHJvdmlkZXJzL01pY3Jvc29mdC5Qb2xpY3lJbnNpZ2h0cy9wb2xpY3lTdGF0ZXMvbGF0ZXN0L3N1bW1hcml6ZT9hcGktdmVyc2lvbj0yMDE4LTA0LTA0JiR0b3A9MTAmJGZyb209MjAxOC0wMy0zMVQwMCUzQTAwJTNBMDBa",
```
Comment (maddieclayton) [inline src/PolicyInsights/PolicyInsights.Test/SessionRecords/Microsoft.Azure.Commands.PolicyInsights.Test.ScenarioTests.PolicyStateSummaryTests/ManagementGroupScope.json]: Please rerecord these test, we need rerecording for a new API version, not just manipulation of the json files.

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by maddieclayton** [comment]
@azuresdkci Add to whitelist

File: src/PolicyInsights/PolicyInsights.Test/SessionRecords/Microsoft.Azure.Commands.PolicyInsights.Test.ScenarioTests.PolicyStateSummaryTests/ManagementGroupScope.json (hunk 1 above)
Comment (j5lim) [reply src/PolicyInsights/PolicyInsights.Test/SessionRecords/Microsoft.Azure.Commands.PolicyInsights.Test.ScenarioTests.PolicyStateSummaryTests/ManagementGroupScope.json]: Done.

File: src/PolicyInsights/PolicyInsights/Cmdlets/GetAzureRmPolicyState.cs [hunk 2]
```diff
@@ -150,6 +149,10 @@ public class GetAzureRmPolicyState : PolicyInsightsCmdletBase
         [ValidateNotNullOrEmpty]
         public string Apply { get; set; }
 
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceScope, Mandatory = false, HelpMessage = ParameterHelpMessages.Expand)]
+        [ValidateNotNullOrEmpty]
+        public string Expand { get; set; }
```
Comment (maddieclayton) [inline src/PolicyInsights/PolicyInsights/Cmdlets/GetAzureRmPolicyState.cs]: Please update help using this doc: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md

Please also link to your design review here: https://github.com/Azure/azure-powershell-cmdlet-review-pr, which is required for any API surface change.

Comment (j5lim) [reply src/PolicyInsights/PolicyInsights/Cmdlets/GetAzureRmPolicyState.cs]: @maddieclayton Both are done.

**Pull request merged**